src/main/scala/io/getclump/Clump.scala

```diff
@@ -143,13 +145,16 @@ private[getclump] class ClumpJoin[A, B](a: Clump[A], b: Clump[B]) extends Clump[
       }
 }
 
-private[getclump] class ClumpCollect[T](list: List[Clump[T]]) extends Clump[List[T]] {
-  val upstream = list
-  val downstream = Future.value(List())
+private[getclump] class ClumpCollect[T, C[_] <: Iterable[_]](clumps: C[Clump[T]])(implicit cbf: CanBuildFrom[C[Clump[T]], T, C[T]]) extends Clump[C[T]] {
+  val upstream = 
+    clumps.toList.asInstanceOf[List[Clump[T]]]
```

Comment:
I couldn't avoid this cast, if you see `scala.concurrent.Future.sequece`, it also uses a cast.